File: solana/programs/token-router/src/state/router_endpoint.rs
use anchor_lang::prelude::*;

#[derive(Debug, AnchorSerialize, AnchorDeserialize, Clone, PartialEq, Eq, InitSpace)]
pub enum MessageProtocol {
    Cctp {
        /// CCTP domain, which is how CCTP registers identifies foreign networks.
        domain: u32,
    },
    Canonical,
}

#[account]
#[derive(Debug, InitSpace)]
/// Foreign emitter account data.
pub struct RouterEndpoint {
    pub bump: u8,

    /// Emitter chain. Cannot equal `1` (Solana's Chain ID).
    pub chain: u16,

    /// Emitter address. Cannot be zero address.
    pub address: [u8; 32],

    /// Specific message protocol used to move assets.
    pub protocol: MessageProtocol,
}

impl RouterEndpoint {
    pub const SEED_PREFIX: &'static [u8] = b"endpoint";
}

File: solana/ts/src/tokenRouter/state/RouterEndpoint.ts
import { ChainId } from "@certusone/wormhole-sdk";
import { PublicKey } from "@solana/web3.js";

export type MessageProtocol = {
    cctp?: { domain: number };
    canonical?: {};
};

export class RouterEndpoint {
    bump: number;
    chain: number;
    address: Array<number>;
    protocol: MessageProtocol;

    constructor(bump: number, chain: number, address: Array<number>, protocol: MessageProtocol) {
        this.bump = bump;
        this.chain = chain;
        this.address = address;
        this.protocol = protocol;
    }

    static address(programId: PublicKey, chain: ChainId) {
        const encodedChain = Buffer.alloc(2);
        encodedChain.writeUInt16BE(chain);
        return PublicKey.findProgramAddressSync(
            [Buffer.from("endpoint"), encodedChain],
            programId
        )[0];
    }
}
